namer/namer.cc

```diff
@@ -1049,6 +1094,44 @@ class SymbolDefiner {
         }
     }
 
+    core::FieldRef insertField(core::MutableContext ctx, const core::FoundField &field) {
+        core::ClassOrModuleRef scope;
+
+        // resolver checks a whole bunch of various error conditions here; we just want to
+        // know where to define the field.
+        if (field.kind == core::FoundField::Kind::Class) {
+            scope = ctx.owner.enclosingClass(ctx);
+        } else {
+            scope = ctx.selfClass();
+        }
+
+        auto existing = scope.data(ctx)->findMember(ctx, field.name);
+        if (existing.exists() && existing.isFieldOrStaticField()) {
+            auto prior = existing.asFieldRef();
+            // There was a previous declaration of this field in this file.  Ignore it;
+            // let resolver deal with issuing the appropriate errors.
+            if (prior.data(ctx)->resultType == core::Types::todo()) {
+                return prior;
+            }
+
+            // We are on the fast path and there was a previous declaration.
+            // TODO(froydnj) apparently incremental resolve can't add new symbols?!
+            // ctx.state.mangleRenameSymbol(existing, existing.data(ctx)->name);
```

Comment:
Consider the case of changing the type of a method on the fast path: we don't mangle rename the symbol, we simply update its type. It should be the same for instance variables unless I misunderstand.

(The ENFORCE that we haven't added new symbols after incremental resolve is something that I think makes sense to keep.)